File: solutions/cpp/crypto-square/1/crypto_square.cpp
#include "crypto_square.h"
#include <regex>
#include <cmath>

using namespace std;

namespace crypto_square {

    cipher::cipher(string plaintext){
        norm_text = "";
        for(string::iterator it=plaintext.begin(); it!=plaintext.end(); ++it){
            char ch = *it;
            if(isalpha(ch))
                norm_text += tolower(ch);
            else if(isdigit(ch))
                norm_text += ch;
        }
        unsigned int len = norm_text.length();
        cols = static_cast<unsigned int>(ceil(sqrt(len)));
        rows = (cols-1)*cols >= len ? cols-1 : cols;
        pad = cols*rows - len;
    }

    string cipher::normalized_cipher_text(){
        string out="";
        unsigned int len = norm_text.length();
        for(unsigned int i=0; i<cols; i++){
            for(unsigned int j=0; j<rows; j++){
                unsigned int idx = j*cols + i;
                if(idx<len) 
                    out += norm_text.at(idx);
            } 
            if(i>=cols-pad) out+=" ";
            if(cols>0 && i<cols-1) out+=" ";
        }
        return out;
    }
        
    

}  // namespace crypto_square

File: solutions/python/crypto-square/1/crypto_square.py
import math

def cipher_text(plain_text):
    clean_text = "".join(filter(str.isalnum, plain_text.lower()))
    length = len(clean_text)
    columns = math.ceil(math.sqrt(length))
    rows = columns
    if columns*(columns-1)>=length:
        rows -=1
    padding = columns*rows - length
    output = ""
    for i in range(0,columns):
        for j in range(0,rows):
            idx = j*columns + i
            if idx<length:
                output += clean_text[idx]
        if i>=columns-padding:
            output += ' '
        if columns>0 and i<columns-1:
            output += ' '
    return output
                